rules/TypeDeclaration/Rector/FunctionLike/AddClosureParamTypeForArrayMapRector.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\TypeDeclaration\Rector\FunctionLike;
+
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\Closure;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Param;
+use PHPStan\Reflection\Native\NativeFunctionReflection;
+use PHPStan\Type\ArrayType;
+use PHPStan\Type\MixedType;
+use PHPStan\Type\Type;
+use PHPStan\Type\UnionType;
+use PHPStan\Type\UnionTypeHelper;
+use Rector\NodeTypeResolver\TypeComparator\TypeComparator;
+use Rector\PHPStanStaticTypeMapper\Enum\TypeKind;
+use Rector\Rector\AbstractRector;
+use Rector\Reflection\ReflectionResolver;
+use Rector\StaticTypeMapper\StaticTypeMapper;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\TypeDeclaration\Rector\FunctionLike\AddClosureParamTypeForArrayMapRector\AddClosureParamTypeForArrayMapRectorTest
+ */
+final class AddClosureParamTypeForArrayMapRector extends AbstractRector
+{
+    public function __construct(
+        private readonly TypeComparator $typeComparator,
+        private readonly StaticTypeMapper $staticTypeMapper,
+        private readonly ReflectionResolver $reflectionResolver,
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Applies type hints to array_map closures',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+array_map(function ($value, $key): string {
+    return $value . $key;
+}, $strings);
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+array_map(function (string $value, int $key): bool {
+    return $value . $key;
+}, $strings);
+CODE_SAMPLE
+                    ,
+                ),
+            ]
+        );
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [FuncCall::class];
+    }
+
+    /**
+     * @param FuncCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($node->isFirstClassCallable()) {
+            return null;
+        }
+
+        if (! $this->isName($node, 'array_map')) {
+            return null;
+        }
+
+        $funcReflection = $this->reflectionResolver->resolveFunctionLikeReflectionFromCall($node);
+
+        if (! $funcReflection instanceof NativeFunctionReflection) {
+            return null;
+        }
+
+        if (! $node->args[0] instanceof Arg || ! $node->args[0]->value instanceof Closure) {
```

Comment:
while undefined index case is almost zero, add `isset()` first, ensure everything ok:

```suggestion
        $args = $node->getArgs();
        if (! isset($args[0]) || ! $args[0]->value instanceof Closure) {
```

as first class callable already checked early above, this usage of `getArgs()` is fine so it always an `Arg`

Reply:
Thank you @samsonasik, I've made the appropriate change